### PR TITLE
Create API endpoint to list repositories Tako is installed on

### LINKED ISSUE #10: Create API endpoint to list repositories Tako is installed on
- [x] `GET` endpoint for retrieving list of repository names
- [x] `DELETE` endpoint for purging cached list of repository names
- [x] Convert existing code to TypeScript
- [x] Error handling
- [x] Unit tests

### PR DESCRIPTION
**Pull request by simonplend**
For #10. Error handling and tests will be added once TypeScript has been removed (#22).

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,12 +1,73 @@
...
+
+			app.log.debug({
+				event: `TAKO_GITHUB_INSTALLATION_REPOSITORIES_${action.toUpperCase()}`,
+				repositories: context.payload[`repositories_${action}`]
+			});
+
+			managedRepos.purgeList();
+			app.log.info({ event: 'TAKO_MANAGED_REPOSITORIES_LIST_PURGED' });
+
+			return Promise.resolve();
```
Comment (simonplend) [inline src/index.ts]: If we don't return a Promise, TypeScript complains - why?

File: src/index.ts [hunk 2]
```diff
@@ -1,12 +1,73 @@
-import { Application } from 'probot';
+import assert from 'assert';
+import express from 'express';
```
Comment (simonplend) [inline src/index.ts]: Do we really need to import `express` and require it as a peer dependency in `package.json` just to reference the `Request` and `Response` types?

**Comment by simonplend** [comment]
A couple of issues to resolve:

- [ ] I disliked the use of `lib` as the `outDir` for TypeScript as it means you can't have any directory called `lib` in your source tree due to the need to have `lib` in your `.gitignore` - I changed it to `dist`, but now you have to do `npm run build` before `npm run dev`, which isn't what we want. I'm not even sure how this was automagically working before when the `outDir` was `lib`?
- [ ] prettier (or possibly `precise-commits`) is for some reason changing tabs to spaces when running as part of the git pre-commit hook, even though it should be respecting the settings in the `.editorconfig` file

**Comment by simonplend** [comment]
@sjparkinson @adambraimbridge As my first stab at something with TypeScript, I'd like to sit down and run through this code with you.

**Comment by sjparkinson** [comment]
Yes, let's do this when you're next in. Good for me to see what's changed. I'll give it a clone in the interim (if ops-cop is quite-ish).

File: .gitignore [hunk 3]
```diff
@@ -4,5 +4,5 @@ npm-debug.log
 .env
 package-lock.json
 coverage
-lib
+dist
```
Comment (adambraimbridge) [inline .gitignore]: It's a moot point now, but I was thinking about this earlier. In cases like this, should we go with default configuration, or make changes base on our own arbitrary, not-so-conventional conventions? 

So to apply that to this case, I'd ask, "in our industry, is it a greater-used convention to use `lib` or `dist`?"

How would we even go about answering that question? If we can't, then should we adopt our conventions, or adopt whatever defaults we're given?

Comment (sjparkinson) [reply .gitignore]: 1. Be consistent
2. If in doubt, refer to the first rule

Ideally `being consistent === the defaults`.

I think defaults are what we should strive for. But we should also accept that sometimes we have deviated, and shouldn't start Yet-Another-Migration by trying to move back to the defaults half-heartedly.

**Pull request merged**